imblearn/utils/testing.py

```diff
@@ -120,3 +125,25 @@ def is_abstract(c):
     # itemgetter is used to ensure the sort does not extend to the 2nd item of
     # the tuple
     return sorted(set(estimators), key=itemgetter(0))
+
+
+@contextmanager
+def warns(expected_warning, match=None):
+    """
```

Comment:
you need to improve the docstring

```python
def warns(expected_warning, match=None):
    """Assert a warning is raised with an optional pattern.

       Assert that a code block/function call warns ``expected_warning``
       and raise a failure exception otherwise. It can be used within a
       context manager ``with``.

        Parameters
        ----------
        expected_warning : Warning
            Warning type.

        match : regex str or None, optional
            The pattern to be checked. By default, no check is done.

        Returns
        -------
        None
       
    """
```

Check the english because it is on the top of the head